// thind-js/index.ts
export class Thind {
    uniqueAttribute: string;
  
    constructor(options: { attribute: string }) {
      // Set the unique attribute from the options or default to 'data-my-el'
      this.uniqueAttribute = options?.attribute || 'data-my-el';
    }
  
    // Deals with single element
    getElement(value: string, parent: HTMLElement = document.body): HTMLElement | null {
      // Use querySelector to find elements with the specified attribute and value
      return parent.querySelector(`[${this.uniqueAttribute}="${value}"]`);
    }
  
    // Deals with multiple elements
    getElements(value: string, parent: HTMLElement): NodeListOf<HTMLElement> {
      // Use querySelectorAll to find elements with the specified attribute and value
      return parent.querySelectorAll(`[${this.uniqueAttribute}="${value}"]`);
    }
  
    // Deals with Webflow forms
    disableFormSubmission(form: HTMLFormElement): void {
      // Remove the w-form class from the form
      form.classList.remove('w-form');
      const parent = form.parentElement as HTMLElement;
      if (parent) {
        // Remove the w-form class from the parent element
        parent.classList.remove('w-form');
      }
      // Add an event listener to prevent default form submission
      form.addEventListener('submit', function(event) {
        event.preventDefault();
      });
    }
  }
